### PR TITLE
Update for random ordering in mod_articles_category

### PR DESCRIPTION
**Pull request by xaddict**
I've added in random ordering in this module since it's something I think should be in the core and it's one of the first things I change in new Joomla installations.

### CODE REVIEW

**Comment by brianteeman** [comment]
Please provide full test instructions otherwise people will just ignore it.

A good starting point would be to state which module it is that you have edited<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6648">issues.joomla.org/joomla-cms/6648</a>.</sub>

**Comment by xaddict** [comment]
modules/mod_articles_category/mod_articles_category.xml

**Comment by xaddict** [comment]
There are no test instructions for previous state as there was no random ordering,

In the new situation one can choose "Random" (if it also gets pulled) in ordering at the backend and the selected articles from mod_articles_category will be shown randomly.

**Comment by xaddict** [comment]
I added the other options into this one. It should now be sufficient to add the "Random" ordering option to the Articles Category module.

File: language/en-GB/en-GB.mod_articles_category.ini [hunk 1]
```diff
@@ -90,8 +90,9 @@ MOD_ARTICLES_CATEGORY_OPTION_ORDERINGFEATURED_VALUE="Featured Articles Order"
 MOD_ARTICLES_CATEGORY_OPTION_RELATIVEDAY_VALUE="Relative Date"
 MOD_ARTICLES_CATEGORY_OPTION_STARTPUBLISHING_VALUE="Start Publishing Date"
 MOD_ARTICLES_CATEGORY_OPTION_FINISHPUBLISHING_VALUE="Finish Publishing Date"
+MOD_ARTICLES_CAREGORY_OPTION_RANDOM_VALUE="Random"
```
Comment (brianteeman) [inline language/en-GB/en-GB.mod_articles_category.ini]: Please correct to say CATEGORY not caRegory

File: modules/mod_articles_category/mod_articles_category.xml [hunk 2]
```diff
@@ -271,6 +271,7 @@
 					<option value="modified">MOD_ARTICLES_CATEGORY_OPTION_MODIFIED_VALUE</option>
 					<option value="publish_up">MOD_ARTICLES_CATEGORY_OPTION_STARTPUBLISHING_VALUE</option>
 					<option value="a.publish_down">MOD_ARTICLES_CATEGORY_OPTION_FINISHPUBLISHING_VALUE</option>
+					<option value="rand()">MOD_ARTICLES_CAREGORY_OPTION_RANDOM_VALUE</option>
```
Comment (brianteeman) [inline modules/mod_articles_category/mod_articles_category.xml]: Please correct to say CATEGORY not caRegory

**Comment by brianteeman** [comment]
Sorry but I am confused. You have added an options called Random to an option 
> "Article Field to order by"
So that definitely isnt the correct place for it
Would it be better to place it in the field below
> "Ordering Direction"
Although that would mean that it was renamed to just Ordering<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6648">issues.joomla.org/joomla-cms/6648</a>.</sub>

**Comment by brianteeman** [comment]
Unfortunately this will not work in Postrges which does not have a rand function<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6648">issues.joomla.org/joomla-cms/6648</a>.</sub>

**Comment by brianteeman** [comment]
@xaddict Thanks for your contribution. The last comment on this comes from myself stating that it cannot be accepted as is because the code doesnt work with Postgres. 

Are you able to update the code so that it will? If not then this will have to be closed<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/6648">issues.joomla.org/joomla-cms/6648</a>.</sub>

**Comment by xaddict** [comment]
You can go ahead and close this. I'm doing this via a custom written module
now which supports random ordering, offset (take 10 items, but only show
the last 5) and splitting (one module shows items 1-3, another one 4-6, the
last shows random items but not 1-6)

**Comment by Bakual** [comment]
Closing per request and since it has open issues anyway.

**Comment by zero-24** [comment]
If there is a willing to make it work on multi database you can use "random" as option value and implement it later with 
```
case 'random' :
	$orderby = JFactory::getDbo()->getQuery(true)->Rand();
	break;
```

**Comment by alikon** [comment]
sorry @xaddict   unable  to send a pr to your branch
nice feature i think , i'll submit a new one, db portable as @zero-24  suggested

**Comment by zero-24** [comment]
:+1:

**Comment by alikon** [comment]
see #8538

**Pull request closed without merging**